src/Freedom/Markup/Common.purs

```diff
@@ -14,115 +9,60 @@ module Freedom.Markup.Common
-
--- | Create `VObject` with passed tag
-tag
-  :: forall f state m
-   . Functor (f state)
-  => MonadRec m
-  => String
-  -> VObject f state m
-tag tag' =
-  { tag: tag'
-  , props: empty
-  , handlers: empty
-  , children: []
-  , didCreate: const $ pure unit
-  , didUpdate: const $ pure unit
-  , didDelete: const $ pure unit
-  }
-
--- | Add children to `VObject`
+-- | Add children.
 kids
-  :: forall f state m
-   . Functor (f state)
-  => MonadRec m
-  => Array (VNode f state)
-  -> VObject f state m
-  -> VObject f state m
-kids children = _ { children = children }
+  :: forall state
+   . Array (VNode state)
+  -> VNode state
+  -> VNode state
+kids children = mapVObject _ { children = children }
 
--- | Add a property to `VObject`
+-- | Add a property.
 prop
-  :: forall f state m
-   . Functor (f state)
-  => MonadRec m
-  => String
+  :: forall state
+   . String
   -> String
-  -> VObject f state m
-  -> VObject f state m
-prop name val obj =
-  obj { props = insert name val obj.props }
+  -> VNode state
+  -> VNode state
+prop name val = mapVObject \vobject ->
+  vobject { props = insert name val vobject.props }
 
--- | Bind an event handler to `VObject`
+-- | Bind an event handler.
 handle
-  :: forall f state m
-   . Functor (f state)
-  => MonadRec m
-  => String
-  -> (Event -> FreeT (f state) m Unit)
-  -> VObject f state m
-  -> VObject f state m
-handle name h obj =
-  obj { handlers = insert name h obj.handlers }
+  :: forall state
+   . String
+  -> (Event -> Operation state -> Effect Unit)
+  -> VNode state
+  -> VNode state
+handle name h = mapVObject \vobject ->
+  vobject { handlers = insert name h vobject.handlers }
 
--- | Bind `didCreate` lifecycle to `VObject`
+-- | Bind `didCreate` lifecycle.
 didCreate
-  :: forall f state m
-   . Functor (f state)
-  => MonadRec m
-  => (Element -> FreeT (f state) m Unit)
-  -> VObject f state m
-  -> VObject f state m
-didCreate h = _ { didCreate = h }
+  :: forall state
+   . (Element -> Operation state -> Effect Unit)
+  -> VNode state
+  -> VNode state
+didCreate h = mapVObject _ { didCreate = h }
 
--- | Bind `didUpdate` lifecycle to `VObject`
+-- | Bind `didUpdate` lifecycle.
 didUpdate
-  :: forall f state m
-   . Functor (f state)
-  => MonadRec m
-  => (Element -> FreeT (f state) m Unit)
-  -> VObject f state m
-  -> VObject f state m
-didUpdate h = _ { didUpdate = h }
+  :: forall state
+   . (Element -> Operation state -> Effect Unit)
+  -> VNode state
+  -> VNode state
+didUpdate h = mapVObject _ { didUpdate = h }
 
--- | Bind `didDelete` lifecycle to `VObject`
+-- | Bind `didDelete` lifecycle.
 didDelete
-  :: forall f state m
-   . Functor (f state)
-  => MonadRec m
-  => (Element -> FreeT (f state) m Unit)
-  -> VObject f state m
-  -> VObject f state m
-didDelete h = _ { didDelete = h }
+  :: forall state
+   . (Element -> Operation state -> Effect Unit)
+  -> VNode state
+  -> VNode state
+didDelete h = mapVObject _ { didDelete = h }
```

Comment:
Move all functions in this module to UI module, because I want to hide VObject and mapVObject.